dio/lib/src/dio/dio_for_native.dart

```diff
@@ -95,7 +97,12 @@ class DioForNative with DioMixin implements Dio {
     // Shouldn't call file.writeAsBytesSync(list, flush: flush),
     // because it can write all bytes by once. Consider that the file is
     // a very big size (up to 1 Gigabytes), it will be expensive in memory.
-    RandomAccessFile raf = file.openSync(mode: FileMode.write);
+    RandomAccessFile raf = file.openSync(
+      mode: fileMode.map(
+        write: () => FileMode.write,
+        append: () => FileMode.append,
```

Comment:
What will happen if the request is not range-supported?

Reply:
it's a thing developer will validate before using this feature. in the package we are only enabling developers to append data to an existing file.

the feature will be useless if there is no range-support but it's a rare case

Reply:
> it's a thing developer will validate before using this feature. in the package we are only enabling developers to append data to an existing file.

Will they? When such features are provided without proper warning or case coverage, developers will have like 50 percent chance to be aware of the edges of the feature, but the user that uses their products will likely have zero chance to know whether their developers have implemented this correctly, so that means we break users experiences.

> the feature will be useless if there is no range-support but it's a rare case

It is common that we live in old systems, and that is what happened with Dio as you may find them in our repo issues.

Reply:
> Will they? When such features are provided without proper warning or case coverage, developers will have like 50 percent chance to be aware of the edges of the feature, but the user that uses their products will likely have zero chance to know whether their developers have implemented this correctly, so that means we break users experiences.

as I said It's not our responsibility to check for range support, we just enabled users to change file access mode